Source/File.System.Stub/DirectoryStub.cs

```diff
@@ -8,5 +8,22 @@ public sealed class DirectoryStub
 		public ICollection<DirectoryStub> Directories { get; set; }
 		public ICollection<FileStub> Files { get; set; }
 		public DirectoryStub ParentDirectory { get; set; }
+
+		public override bool Equals(object obj)
+		{
+			var stub = obj as DirectoryStub;

```

Comment:
You're not checking ParentDirectory here

Reply:
If I take parent dir to equals we'll generate StackOverflowException. :) If we want to do it, we should add some possibility to get parent equality without involving children dirs, or checking only folder names.